pkg/tableprinter/table.go

```diff
@@ -0,0 +1,238 @@
+// Package tableprinter facilitates rendering column-formatted data to a terminal and TSV-formatted data to
+// a script or a file.
+package tableprinter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/mattn/go-runewidth"
+)
+
+type fieldOption func(*tableField)
+
+type TablePrinter interface {
+	AddField(string, ...fieldOption)
+	EndRow()
+	Render() error
+}
+
+// WithTruncate sets the truncation function for the field. Pass nil to disable truncation for this value.
```

Comment:
If no `WithTruncate` function is specified for a field will the the field be truncated as necessary to fit the output width?

Reply:
That's a good question! Every field is automatically truncated by default. I will try to point this out in the docs